### PR TITLE
Hapi +  PostgreSQL Auto-Instrumentation Example

### PR DESCRIPTION
**Pull request by sm7x**

### CODE REVIEW

File: signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+
+
+
```
Comment (rmfitzpatrick) [inline signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md]: nit: do you mind removing extraneous whitespace?  Also I think a single blank line between segments in js files is an accepted convention (used in both google and airbnb guides).

File: signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  $ npm run postgres
+  $ # install app requirements
+  $ npm install
+  $ # Run the server from one shell session:
+  $ npm start
+
+  ++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+        Welcome to ContactKeeper.
+        The server is listening on http://localhost:.8000
```
Comment (rmfitzpatrick) [inline signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md]: `.8000`

File: signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# Hapi.js + PostgreSQL Auto-Instrumentation Example
+
+This is an example of automatically producing distributed traces using the
+[SignalFx Tracing Library for JavaScript](https://github.com/signalfx/signalfx-nodejs-tracing).
+Please examine the instrumented [client](./client.js) and [server](./server.js) for
+some basic patterns in accessing the instrumentations of a [http](https://nodejs.org/api/http.html)
+client and a [Hapi](https://hapijs.com) server using a [node-postgres](https://github.com/brianc/node-postgres) client.
+
+In ths example, we have a simple contact application named
```
Comment (rmfitzpatrick) [inline signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md]: `this`

Comment (sm7x) [reply signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md]: Thank you!

File: signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/contactKeeper/config.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const serverPort = 8000;
+const serverHost = 'localhost';
+
+
+module.exports = {
+  serverHost,
+  serverPort,
+  'serverUrl': `http://localhost:${serverPort}`,
```
Comment (mpetazzoni) [inline signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/contactKeeper/config.js]: Shouldn't this be `http://${serverHost}:${serverPort}` ?

File: signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+To run this example locally and send traces to your available Smart Agent or Gateway,
+ from this directory do the following:
+
+In one terminal:
+```bash
+  $ # start up the postgres server
+  $ npm run postgres
+  $ # install app requirements
+  $ npm install
```
Comment (mpetazzoni) [inline signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md]: Fix indent

File: signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  $ # Run the server from one shell session:
+  $ npm start
```
Comment (mpetazzoni) [inline signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/README.md]: Fix indent

File: signalfx-tracing/signalfx-nodejs-tracing/README.md [hunk 7]
```diff
@@ -13,6 +13,7 @@ instrumentation configuration options, please see the
 ## Libraries and Framework Examples
 
 - [Express](./express)
+- [Hapi + Postgres](./hapi_pg)
```
Comment (mpetazzoni) [inline signalfx-tracing/signalfx-nodejs-tracing/README.md]: I would list this as an Hapi example, and have another line about Postgres that points people back at the Hapi example, to be clear about the fact that those don't necessarily have to be used together when instrumenting.

File: signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/client.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+// Initialize the tracer and initiate auto-instrumentation
```
Comment (mpetazzoni) [inline signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/client.js]: This comment doesn't really match what happens in the code below, and ends up being confusing to the reader.

File: signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/client.js [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+// of all supported libraries and frameworks.  In this client example,
+// we instrumented the http module used by the contactKeeper client module.
+// const tracer = require('./contactKeeper/tracer');
+
+// Note that importing other modules should occur after init()
```
Comment (mpetazzoni) [inline signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/client.js]: Again here, there is no call to `init()` in sight, making this quite confusing

File: signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/package.json [hunk 10]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "hapi_pg",
+  "version": "1.0.0",
+  "description": "A contact keeping application illustrating SignalFX Hapi.js and PostgreSQL Auto-Instrumentation ",
```
Comment (mpetazzoni) [inline signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/package.json]: s/SignalFX/SignalFx/

File: signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/client.js (hunk 9 above)
Comment (sm7x) [reply signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/client.js]: Thanks! Would changing it to:
>`Note that importing other modules should occur after importing the tracer
> to ensure their supported dependencies are auto-instrumented.` 

be less confusing?

Comment (sm7x) [reply signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/client.js]: I just realised that there is no need to import the tracer in this file. Tracing still is auto-instrumented. Hope I'm not overlooking something?

Comment (mpetazzoni) [reply signalfx-tracing/signalfx-nodejs-tracing/hapi_pg/client.js]: I'd say just take it out

**Comment by mpetazzoni** [comment]
@sm7x What's the hold up here? This has been approved for 6 weeks.

**Comment by sm7x** [comment]
> @sm7x What's the hold up here? This has been approved for 6 weeks.

**Comment by sm7x** [comment]
> @sm7x What's the hold up here? This has been approved for 6 weeks.
I apologize. I wanted to give it a once over, and I got caught up with new projects. Will complete the once over and merge.

**Pull request merged**